cms/models.py

```diff
@@ -2,16 +2,68 @@
 Page models for the CMS
 """
 from django.db import models
+from django.utils.text import slugify
 
 from wagtail.admin.edit_handlers import FieldPanel, StreamFieldPanel
 from wagtail.core import blocks
-from wagtail.core.models import Page
+from wagtail.core.models import Page, Orderable
 from wagtail.core.fields import RichTextField, StreamField
 from wagtail.core.blocks import RawHTMLBlock
 from wagtail.images.models import Image
 from wagtail.images.blocks import ImageChooserBlock
 
 
+class CourseProgramChildPage(Page):
+    """
+    Abstract page representing a child of Course/Program Page
+    """
+    class Meta:
+        abstract = True
+
+    parent_page_types = ['CoursePage', 'ProgramPage']
+
+
+class LearningOutComesPage(CourseProgramChildPage):
+    """
+    Learning outcomes page for learning benefits.
+    """
+    subpage_types = []
+    heading = RichTextField(
+        blank=False,
+        help_text="Heading highlighting the learning outcomes generally.",
+    )
+    sub_heading = models.CharField(
+        max_length=250,
+        null=True,
+        blank=False,
+        help_text="Sub heading for learning outcomes.",
+    )
+
+    learning_outcomes = StreamField([
+        ("outcome", blocks.TextBlock(icon="plus"))
+    ],
+        blank=False,
+        help_text="Detail about What you'll learn as learning outcome.",
+    )
+
+    content_panels = [
+        FieldPanel("heading"),
+        FieldPanel("sub_heading"),
+        StreamFieldPanel("learning_outcomes"),
+    ]
+
+    def save(self, *args, **kwargs):
+        new_title = "What you will learn?"
+        self.title = new_title
+        super(LearningOutComesPage, self).save()
+
+    @classmethod
+    def can_create_at(cls, parent):
+        # You can only create one of these page under course / program.
+        return super(LearningOutComesPage, cls).can_create_at(parent) \
+               and parent.get_children().type(LearningOutComesPage).count() == 0
```

Comment:
Wouldn't it be the same for all types of sub-pages on a course/program? I mean, we'll have one page per section, and given the design a section does not repeat, which means all types like `LearningOutcomesPage` and `WhatYouWillLearnPage` etc will always have a maximum count of 1 per `CoursePage` or `ProgramPage`. Does it make sense to move this into `CourseProgramChildPage`?

Reply:
we need to check that as I am using the query in which pointing to `LearningOutComesPage` only. 
`parent.get_children().type(LearningOutComesPage).count`